test/subshell/test_runner_test.go

```diff
@@ -32,6 +32,20 @@ func TestMockingRunner(t *testing.T) {
 		must.EqOp(t, "foo called with: bar", res)
 	})
 
+	t.Run("MockCommitMessage", func(t *testing.T) {
+		t.Parallel()
+		runner := subshell.TestRunner{
+			WorkingDir: t.TempDir(),
+			HomeDir:    t.TempDir(),
+			BinDir:     filepath.Join(t.TempDir(), "bin"),
+		}
```

Comment:
You could save a few roundtrips to the filesystem during and then again after the test (to clean up the created temp dirs).

```go
dir := t.TempDir()
runner := subshell.TestRunner{
	WorkingDir: dir,
	HomeDir:    dir,
	BinDir:     filepath.Join(dir, "bin"),
}
```

Reply:
Didn't realise `t.TempDir()` returns a new tempdir every time. Thanks!